### ISSUE TITLE
Don't enforce superfluous-parens for default function values

### ISSUE THREAD

**Issue by skirpichev** [PyCQA/pylint#8160]
### Bug description

Probably, using complex expressions as defaults is a kind of a bad practice, but the language *does* support this.  I'm suggesting to relax the checker for such examples:
```
$ cat -n a.py 
     1  TEST = 1
     2  
     3  def foo1(spam=(TEST == 0)):  # superfluous-parens
     4      return spam
     5  
     6  
     7  def foo2(spam=TEST == 0):  # unreadable
     8      return spam
     9  
    10  
    11  def foo3(spam= TEST == 0):  # only slightly better, but it's not PEP 8
    12      return spam
    13  
    14  
    15  def foo4(spam= TEST == ((3 + 2))):  # but here we want C0325!
    16      return spam
```

(Last case, IIUIC, is #5361.)

### Command used

```shell
pylint -s n --disable=all --enable=C0325 a.py
```

### Pylint output

```shell
************* Module a
a.py:3:0: C0325: Unnecessary parens after '=' keyword (superfluous-parens)
```

### Expected behavior

No error.

### Pylint version

```shell
pylint 2.16.0
astroid 2.14.1
Python 3.11.1+ (heads/3.11:ba88628808, Jan 14 2023, 08:40:06) [GCC 10.2.1 20210110]
```

**Comment by Pierre-Sassoulas** [comment]
I think this will feel like a false negative if we do, the warning is accurate the parenthesis can be removed or the warning disabled. We might add an option though.

**Comment by skirpichev** [comment]
>  the warning is accurate

UPD: Yes, but "spam=TEST == 0" looks less readable then the original variant.  I think this holds for any default value if it's a BinOp.

**Comment by tamird** [comment]
It would also be good to add an option to allow superfluous parens preceding negation.
```diff
-        if not (MIN <= timeout_seconds <= MAX): # before appeasing superfluous-parens, readable
+        if not MIN <= timeout_seconds <= MAX: # after appeasing superfluous-parens, unreadable
```

I've shortened the names of the constants `MIN` and `MAX`; this becomes even less readable as the length of the expression grows.
